core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -291,8 +292,12 @@ object SparkSubmit {
       } else {
         Nil
       }
+
+    val ivySettings = Option(args.ivySettingsFile).map(SparkSubmitUtils.loadIvySettings).getOrElse(
```

Comment:
So basically it's either / or. Is there any way to reconcile the two? (e.g. load the file and add repositories explicitly defined in the command line). This would allow one to override the settings file in the default Spark conf and still be able to add extra repositories.